app/src/lib/survey.ts

```diff
@@ -0,0 +1,43 @@
+import {ProjectUIModel} from '@faims3/data-model/build/src/types';
+
+/**
+ * Retrieves the keys of fields that are marked as required from the given project UI model.
+ *
+ * @param {ProjectUIModel} fields - An object representing the project's UI model fields, where each field contains component parameters.
+ * @returns {string[]} An array of keys representing the fields that are marked as required.
+ */
+export const requiredFields = (fields: ProjectUIModel): string[] =>
+  Object.entries(fields)
+    .filter(([, value]) => value['component-parameters'].required)
+    .map(([key]) => key);
+
+/**
+ * Determines whether a given value is considered "empty".
+ * A value is considered empty if it is an empty array, an empty string, or explicitly set to null.
+ *
+ * @param {any} value - The value to be checked. It can be of any type.
+ * @returns {boolean} - Returns true if the value is considered empty, otherwise false.
+ */
+const emptyValue = (value: any): boolean => {
```

Comment:
Boolean fields default to 'false' - this means they are reported as complete even when user has never interacted with them. This is out of scope of this PR though.